.github/workflows/process_new_pr.yml

```diff
@@ -0,0 +1,127 @@
+            const repo = '${{ github.event.repository.name }}';
+            
+            try {
+              const { data } = await github.rest.search.issuesAndPullRequests({
+                q: `repo:${owner}/${repo} type:pr is:merged author:${author}`,
+              });
+              return data.total_count === 0;
+            } catch (error) {
+              console.error('Error checking contributor status:', error);
+              return false;
+            }
+      - name: Check for Assigned Reviewers and Teams
+        id: check-reviewers
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const pr = context.payload.pull_request;
+            const hasReviewers = pr.requested_reviewers?.length > 0;
+            const hasTeams = pr.requested_teams?.length > 0;
+            core.setOutput('has_reviewers', hasReviewers);
+            core.setOutput('has_teams', hasTeams);
+  manage_reviewers_and_comments:
+    name: Assign Reviewers and Post Comments
+    needs: check_contributor_and_reviewers
+    runs-on: ubuntu-22.04
+    permissions:
+      pull-requests: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Merge develop and set up dependencies
+        uses: ./.github/actions/merge-develop-and-setup-python
+
+      - name: Assign PR Reviewers as Assignees
+        if: needs.check_contributor_and_reviewers.outputs.has_reviewers == 'true'
+        uses: actions/github-script@v7
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const owner = '${{ needs.check_contributor_and_reviewers.outputs.owner }}';
+            const repo = '${{ needs.check_contributor_and_reviewers.outputs.repo }}';
+            const pr_number = parseInt('${{ needs.check_contributor_and_reviewers.outputs.pr_number }}');
+            
+            try {
+              const { data: pr } = await github.rest.pulls.get({
+                owner,
+                repo,
+                pull_number: pr_number
+              });
+              
+              const reviewers = pr.requested_reviewers.map(r => r.login);
+              
+              if (reviewers.length > 0) {
+                await github.rest.issues.addAssignees({
+                  owner,
+                  repo,
+                  issue_number: pr_number,
+                  assignees: reviewers
+                });
+                console.log(`Assigned reviewers: ${reviewers.join(", ")}`);
+              }
+            } catch (error) {
+              console.error('Error assigning reviewers:', error);
+              throw error;
+            }
+      - name: Notify Team if No Reviewers are Assigned
+        if: needs.check_contributor_and_reviewers.outputs.has_reviewers == 'false' && needs.check_contributor_and_reviewers.outputs.has_teams == 'false'
```

Comment:
It looks like if there are reviews requested from a team but no users, we do nothing, which doesn't seem right since that would leave no one assigned (we usually have a user from a team get requested for review, but this isn't guaranteed to happen, e.g. if the team doesn't have any active reviewers)